scrabble.js

```diff
@@ -1,16 +1,130 @@
+// let pry = require('pryjs');
+
+const scores = {
+  a: 1, e: 1, i: 1, o: 1, u: 1, l: 1, n: 1, r: 1, s: 1,
+  t: 1, d: 2, g: 2, b: 3, c: 3, m: 3, p: 3, f: 4, h: 4, v: 4, w: 4, y: 4, k: 5, j: 8, x: 8, q: 10, z: 10
+}
 
 const Scrabble = {
+
   score(word) {
+    if (word === '') {
+      throw 'Must play a word with at least one letter.';
+    }
+    if (word.length > 7) {
+      throw 'Cannot accept word longer than seven letters.';
+    }
+    word = word.toLowerCase();
+    let wordArray = word.split('');
+    let totalScore = 0;
+    wordArray.forEach(function(char) {
+      if (scores[char] == undefined) {
+        throw `${char} is not a valid character.`
+      }
+      let charValue = scores[char];
+      totalScore += charValue;
+    });
+    if (word.length == 7) {
+      totalScore += 50;
+    }
+    return totalScore;
+  }, // score
 
-  },
   highestScoreFrom(arrayOfWords) {
+    if (!Array.isArray(arrayOfWords) || arrayOfWords.length == 0) {
+      throw 'Must enter a valid array of words.'
+    }
+
+    let wordsAndScores = {};
+
+    let highestScore = this.score(arrayOfWords[0]);
+    for (let word of arrayOfWords) {
+      let score = this.score(word);
+      wordsAndScores[word] = score;
```

Comment:
This function ends up being fairly long and complex - is there some way you could clarify it, possibly using a helper function?